memcached.c

```diff
@@ -3643,6 +3643,8 @@ static int server_sockets(int port, enum network_transport transport,
             fprintf(stderr, "Failed to allocate memory for parsing server interface string\n");
             return 1;
         }
+        // If we encounter any failure, preserve the first errno for the caller.
```

Comment:
[Note] Preserving the first error here since the following case can occur:
Considering 2 interfaces:
./memcached  -l <IF1>,<IF2>

If memcached fails to bind to <IF1>, but succeeds binding to <IF2>, the error message will look like so:
"failed to listen on one of interface(s) <IF1>,<IF2>: Success"

If we instead preserve the first error seen, we'll always see a valid error:
"failed to listen on one of interface(s) <IF1>,<IF2>: address already in use"